client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx

```diff
@@ -207,6 +93,62 @@ const JetpackComMasterbar: React.FC< Props > = ( { pathname } ) => {
 		} );
 	}, [] );
 
+	const toggleMenuItem = ( btn: HTMLAnchorElement, menu: HTMLUListElement ) => {
+		const expanded = btn.getAttribute( 'aria-expanded' ) === 'true' || false;
+		btn.setAttribute( 'aria-expanded', String( ! expanded ) );
+
+		menu.hidden = ! menu.hidden;
+	};
+
+	const collapseExpandedMenu = () => {
+		const expandedBtn = document.querySelector(
+			'.js-menu-btn[aria-expanded="true"]'
+		) as HTMLAnchorElement;
+
+		if ( expandedBtn ) {
+			const menu = expandedBtn?.parentNode?.querySelector( 'js-menu' ) as HTMLUListElement;
+
+			if ( menu ) {
+				toggleMenuItem( expandedBtn, menu );
+			}
+		}
+	};
+
+	const onMenuBtnClick = ( e: React.MouseEvent< HTMLAnchorElement > ) => {
+		e.preventDefault();
+		const btn = e.currentTarget;
+		const menu = btn.parentNode?.querySelector( '.js-menu' ) as HTMLUListElement;
+
+		if ( btn.getAttribute( 'aria-expanded' ) === 'false' ) {
+			collapseExpandedMenu();
+		}
+
+		toggleMenuItem( btn, menu );
+	};
+
+	const getBundleIcons = ( bundle: string ) => {
+		// Using a soft match in case the menu item gets deleted and recreated in wp-admin
+		// causing the name to change to `complete-2` or something similar.
+		if ( bundle.includes( 'complete' ) ) {
+			return [
+				<BackupIcon />,
+				<AntispamIcon />,
+				<ScanIcon />,
+				<SearchIcon />,
+				<SocialIcon />,
+				<VideopressIcon />,
+				<CRMIcon />,
+				<BoostIcon />,
+			];
+		}
+
+		if ( bundle.includes( 'security' ) ) {
+			return [ <BackupIcon />, <AntispamIcon />, <ScanIcon /> ];
+		}
+
+		return [];
+	};
+
```

Comment:
In general, the component code and styles look good when tested - similar to jetpack.com diff.

However, I'm wondering if we could improve the readability of this code. React gives us a nice possibility to modularize view components, and here we end up with a large nested structure that has many indents.

We could at least make a small improvement by setting a new pattern in the new code that is added here - I don't mean the big refactor of the whole file at once. Making such a big change to the menu is a good opportunity to do that.

Reply:
I have a separate task to move some of this to components and clean up the code 😄